### ISSUE TITLE
Space is forced at 5th position in message

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#496]
In GitLab by @xZise on Jun 1, 2016, 11:44

As mentioned in #88, it forces a space in the output after the fourth character. So if the message is `BUG>X<-- Space forced there`, the actual output is `…:1:2: BUG> <-- Space forced there`. Now you suggested a sample code, but to be honest I'm not exactly sure how to properly trim it down, so here is the class:

```
class Bug141(object):

    version = '1'
    name = 'flake8-bug141'

    def __init__(self, tree, filename):
        pass

    def run(self):
        yield 1, 1, "BUG>X<-- Space forced there", type(self)
```

This also affects how `--ignore` works:
* `--ignore "BUG> "` works, which might be because it is trimming the text, in which case it is not related to the issue above
* `--ignore "BUG> <"` works not
* `--ignore "BUG>_<"` works not too

Apart from that, it should be documented how the error code should be formatted. Some assume that it's always `[A-Z]\d\d\d` while others assume it's always `[A-Z0-9]{4}` while it could be actually anything apart from a space (to separate code from the message).

**Comment by asottile** [comment]
In GitLab by @xZise on Jul 26, 2016, 08:40

This has been fixed with at least 3.0.1 but probably already with 3.0.0. So if you want to fix it, it probably needs to be in Flake 2.x (if you plan to support it):

```
$ echo 'this_should_be_linted = "double quote string"' | flake8 --select=Q - ; flake8 --version
stdin:1:24: Q000111 Remove bad quotes.
3.0.1 (flake8_quotes: 0.7.0, flake8-string-format: 0.2.2, pycodestyle: 2.0.0, mccabe: 0.5.0, pyflakes: 1.2.3) CPython 3.5.1 on Linux
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jul 26, 2016, 08:41

Milestone removed

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jul 26, 2016, 08:43

I'm 90% certain this is actually a problem in pep8/pycodestyle where it truncates the error code like this. I'm not certain there is anything to do in 2.x for this sadly :(. I'll double check but I don't consider this a high priority since I don't intend to support 2.6 now that 3.0 is out. The `stable/2.6` branch was there for 2.6 bug fixes while 3.0 was being finalized.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jul 29, 2016, 17:56

Just so you're aware, the problem for the 2.x branch of Flake8 is https://github.com/PyCQA/pycodestyle/blob/master/pycodestyle.py#L1775

I think we could probably reimplement that for the `BaseQReport` in 2.6 but we'd also have to do it for plain (non-multiprocessing reports) and that would be a whole other class.

If you want to put in that work, I'll happily merge it and cut another 2.6.x release to make the transition to 3.0 easier, but I'm kind of burning out on this project now. I'm sorry. :weary:
